### PR TITLE
Refactors pallet-nfts to use fungible traits 

### PR DESCRIPTION
**Pull request by jsidorenko**
Part of https://github.com/paritytech/polkadot-sdk/issues/226

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3890239

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3897980

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3898270

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3898373

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3900825

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3901265

File: substrate/frame/nfts/src/features/transfer.rs [hunk 1]
```diff
@@ -116,49 +119,53 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
 			// Move the deposit to the new owner.
-			T::Currency::repatriate_reserved(
+			T::Currency::transfer_on_hold(
+				&HoldReason::CollectionOwnerAggregatedDeposit.into(),
 				&details.owner,
-				&owner,
+				&new_owner,
 				details.owner_deposit,
-				Reserved,
+				BestEffort,
```
Comment (gilescope) [inline substrate/frame/nfts/src/features/transfer.rs:151]: Shoud this be Precise? I'd think if we couldn't transfer all the deposit we'd prefer to fail the transaction?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3952570

**Pull request closed without merging**